File: packages/twilio_flutter_conversations/ios/Classes/Bridge/MessageMediaMapping.swift
import Foundation
import TwilioConversationsClient

enum MessageMediaMapping {
  static func mapMedia(_ media: Media) -> MediaAttachmentDto {
    return MediaAttachmentDto(
      sid: media.sid ?? "",
      contentType: media.contentType ?? "application/octet-stream",
      filename: media.filename ?? "",
      sizeBytes: Int64(media.size)
    )
  }

  static func enrichMessageDto(
    base: MessageDto,
    message: TCHMessage
  ) -> MessageDto {
    let mediaItems = message.attachedMedia
    if mediaItems.isEmpty {
      return base
    }

    let attachments = mediaItems.map { mapMedia($0) }

    return MessageDto(
      sid: base.sid,
      conversationSid: base.conversationSid,
      author: base.author,
      body: base.body,
      messageIndex: base.messageIndex,
      dateCreatedEpochMs: base.dateCreatedEpochMs,
      contentType: .media,
      mediaAttachments: attachments,
      attributesJson: base.attributesJson
    )
  }
}

File: packages/twilio_flutter_conversations/android/src/main/kotlin/com/twilioflutter/conversations/bridge/MessageMediaMapping.kt
package com.twilioflutter.conversations.bridge

import com.twilio.conversations.Media
import com.twilio.conversations.Message
import com.twilioflutter.conversations.pigeon.MediaAttachmentDto
import com.twilioflutter.conversations.pigeon.MessageContentType
import com.twilioflutter.conversations.pigeon.MessageDto

/** Maps Twilio [Message] values including media attachments. */
object MessageMediaMapping {
  fun mapMedia(media: Media): MediaAttachmentDto {
    return MediaAttachmentDto(
      sid = media.sid,
      contentType = media.contentType ?: "application/octet-stream",
      filename = media.filename ?: "",
      sizeBytes = media.size,
    )
  }

  fun enrichMessageDto(
    base: MessageDto,
    message: Message,
  ): MessageDto {
    val attachments =
      message.attachedMedia?.map { mapMedia(it) } ?: emptyList()
    if (attachments.isEmpty()) {
      return base
    }
    return MessageDto(
      sid = base.sid,
      conversationSid = base.conversationSid,
      author = base.author,
      body = base.body,
      messageIndex = base.messageIndex,
      dateCreatedEpochMs = base.dateCreatedEpochMs,
      contentType = MessageContentType.MEDIA,
      mediaAttachments = attachments,
      attributesJson = base.attributesJson,
    )
  }
}
